WordPress/Classes/ViewRelated/Post/AbstractPostListViewController.swift

```diff
@@ -111,6 +106,50 @@ class AbstractPostListViewController : UIViewController, WPContentSyncHelperDele
     override func viewWillAppear(animated: Bool) {
         super.viewWillAppear(animated)
         refreshResults()
+
+        registerForKeyboardNotifications()
+    }
+
+    private func registerForKeyboardNotifications() {
+        NSNotificationCenter.defaultCenter().addObserver(self, selector: #selector(keyboardDidShow(_:)), name: UIKeyboardDidShowNotification, object: nil)
+        NSNotificationCenter.defaultCenter().addObserver(self, selector: #selector(keyboardDidHide(_:)), name: UIKeyboardDidHideNotification, object: nil)
+    }
+
+    private func unregisterForKeyboardNotifications() {
+        NSNotificationCenter.defaultCenter().removeObserver(self, name: UIKeyboardDidShowNotification, object: nil)
+        NSNotificationCenter.defaultCenter().removeObserver(self, name: UIKeyboardDidHideNotification, object: nil)
+    }
+
+    @objc private func keyboardDidShow(notification: NSNotification) {
+        let keyboardFrame = localKeyboardFrameFromNotification(notification)
+        let keyboardHeight = CGRectGetMaxY(tableView.frame) - keyboardFrame.origin.y
+
+        tableView.contentInset.top = topLayoutGuide.length
+        tableView.contentInset.bottom = keyboardHeight
+        tableView.scrollIndicatorInsets.top = searchBarHeight
+        tableView.scrollIndicatorInsets.bottom = keyboardHeight
+    }
+
+    @objc private func keyboardDidHide(notification: NSNotification) {
+        tableView.contentInset.top = topLayoutGuide.length
+        tableView.contentInset.bottom = 0
+        tableView.scrollIndicatorInsets.top = searchBarHeight
+        tableView.scrollIndicatorInsets.bottom = 0
+    }
+
+    private var searchBarHeight: CGFloat {
+        return CGRectGetHeight(searchController.searchBar.bounds) + topLayoutGuide.length
+    }
+
+    private func localKeyboardFrameFromNotification(notification: NSNotification) -> CGRect {
+        guard let keyboardFrame = (notification.userInfo?[UIKeyboardFrameEndUserInfoKey] as? NSValue)?.CGRectValue(),
+            let window = view.window else {
+                return .zero
+        }
+
+        var convertedFrame = window.convertRect(keyboardFrame, fromWindow: nil)
+        convertedFrame = view.convertRect(convertedFrame, fromView: nil)
```

Comment:
Yep, you're right. To be honest this is code I've reused in the past and I _think_ the straight conversion from the window didn't work due to orientation issues or something... Simply doing `view.convertRect(convertedFrame, fromView: nil)` seems to work fine though (passing `nil` uses the window). Thanks!